<launch>

	<!-- VIDEO SOURCE -->
	<include file="$(find-pkg-share ros_deep_learning)/launch/video_source.ros2.launch"/>

	<!-- DETECTNET -->
	<arg name="model_name" default="resnet18-body"/>
	<arg name="model_path" default=""/>
	<arg name="prototxt_path" default=""/>
	<arg name="input_blob" default=""/>
	<arg name="output_cmap" default=""/>
	<arg name="output_paf" default=""/>
	<arg name="overlay_flags" default="OVERLAY_DEFAULT"/>
	<arg name="keypoint_scale" default="0.0052"/>
	<arg name="link_scale" default="0.0013" />
	<arg name="mean_pixel_value" default="0.0"/>
	<arg name="threshold" default="0.5"/>

	<node pkg="ros_deep_learning" exec="posenet" output="screen">
		<remap from="/posenet/image_in" to="/video_source/raw"/>
		<param name="model_name" value="$(var model_name)"/>
		<param name="model_path" value="$(var model_path)"/>
		<param name="prototxt_path" value="$(var prototxt_path)"/>
		<param name="input_blob" value="$(var input_blob)"/>
		<param name="output_cmap" value="$(var output_cmap)"/>
		<param name="output_paf" value="$(var output_paf)"/>
		<param name="overlay_flags" value="$(var overlay_flags)"/>
		<param name="keypoint_scale" value="$(var keypoint_scale)"/>
		<param name="link_scale" value="$(var link_scale)"/>
		<param name="mean_pixel_value" value="$(var mean_pixel_value)"/>
		<param name="threshold" value="$(var threshold)"/>
	</node>

	<!-- VIDEO OUTPUT -->
	<include file="$(find-pkg-share ros_deep_learning)/launch/video_output.ros2.launch">
		<arg name="topic" value="/posenet/overlay"/>
	</include>

</launch>
